internal/server/auth/middleware.go

```diff
@@ -137,6 +138,45 @@ func UnaryInterceptor(logger *zap.Logger, authenticator Authenticator, o ...cont
 	}
 }
 
+// EmailMatchingInterceptor is a grpc.UnaryServerInterceptor only used in the case where the user is using OIDC
+// and wants to whitelist a group of users issuing operations against the Flipt server.
+func EmailMatchingInterceptor(logger *zap.Logger, emailMatches []string) grpc.UnaryServerInterceptor {
+	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
+		auth := GetAuthenticationFrom(ctx)
+
+		if auth == nil {
+			return handler(ctx, req)
+		}
+
+		email, ok := auth.Metadata["io.flipt.auth.oidc.email"]
+		if !ok {
+			logger.Debug("no email provided but required for auth")
+			return ctx, errUnauthenticated
+		}
+
+		matched := false
+
+		for _, em := range emailMatches {
+			if matched {
+				break
+			}
+
+			// We have validated the regex's are valid ahead of time, therefore it is safe to use the MustCompile method
+			// here.
+			rgx := regexp.MustCompile(em)
+
+			matched = matched || rgx.MatchString(email)
```

Comment:
Just a 💅 terse suggestion (assuming you're iterating over a list of precompiled regexp):
```suggestion
			if matched = rgx.MatchString(email); matched {
				break
			}
```